corehq/apps/locations/permissions.py

```diff
@@ -43,11 +43,25 @@ def _inner(request, domain, *args, **kwargs):
     return locations_access_required(_inner)
 
 
+def get_user_location(user, domain):
+    if user.is_commcare_user():
+        return user.location
+    else:
+        return user.get_location(sql_location.domain)
+
+
+def get_user_sql_location(user, domain):
+    if user.is_commcare_user():
+        return user.sql_location
+    else:
+        return user.get_sql_ocation(sql_location.domain)
```

Comment:
thanks, i had tested with a mobile worker and an admin web user so didn't catch this. i now retested with a read-only web user after updating